### PR TITLE
Add feedback for long press on iOS 

### LINKED ISSUE #148391: Feedback.forLongPress does not provide feedback on iOS
### Steps to reproduce

Execute Feedback.forLongPress on an iOS device.
Or run the example on iOS and long press the screen.
Notice that it prints "Long-press started" to the console, but there is no feedback.

### Expected results

Should provide haptic feedback. The documentation says "On iOS this is a no-op as that platform usually doesn't provide feedback for long presses.", however this is wrong from my point of view.
iOS gives feedback for long presses, for example when long pressing on a photo in the Photos app, or long pressing on a message in iMessage. The feedback includes a "tap" Sound, when sounds are enabled, and haptic feedback similar to `HapticFeedback.heavyImpact()`. <br>
I am not an iOS guy, so I would appreciate it if an iOS guy can provide his opinion on this topic.

### Actual results

Does not provide haptic feedback and sound.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: GestureDetector(
        onLongPressStart: (_) {
          print("Long-press started");
          Feedback.forLongPress(context);
          //HapticFeedback.heavyImpact();
        },
        child: Container(
          color: Colors.white,
          child: const Center(child: Text("Long-press me")),
        ),
      ),
    );
  }
}

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel stable, 3.22.0, on macOS 14.1.1 23B81 darwin-arm64, locale en-DE)
    • Flutter version 3.22.0 on channel stable at /Users/tobias/dev/tools/flutter
    ! Warning: `flutter` on your path resolves to /Users/tobias/Dev/Tools/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at
      /Users/tobias/dev/tools/flutter. Consider adding /Users/tobias/dev/tools/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.2.2/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/tobias/dev/tools/flutter. Consider adding /Users/tobias/dev/tools/flutter/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (6 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/tobias/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/tobias/Library/Android/sdk
    • Java binary at: /Users/tobias/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.3)
    • Android Studio at /Users/tobias/Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)

[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.1)
    • IntelliJ at /Users/tobias/Applications/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 79.0.3
    • Dart plugin version 241.15989.9

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (5 available)
    • Pixel 7 (mobile)                • 2B181FDH2001W8                           • android-arm64  • Android 14 (API 34)
    • iPhone X Testgerät (mobile)     • c6337498544ef0daff2c5057749efcdd8118d350 • ios            • iOS 16.7.7 20H330
    • macOS (desktop)                 • macos                                    • darwin-arm64   • macOS 14.1.1 23B81 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                    • darwin         • macOS 14.1.1 23B81 darwin-arm64
    • Chrome (web)                    • chrome                                   • web-javascript • Google Chrome 124.0.6367.203

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by victorsanni**
Adds the click system sound and heavy-impact haptic feedback to iOS on long presses.

Fixes #148391

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/feedback.dart [hunk 1]
```diff
@@ -127,8 +127,9 @@ abstract final class Feedback {
 
   /// Provides platform-specific feedback for a long press.
   ///
-  /// On Android the platform-typical vibration is triggered. On iOS this is a
-  /// no-op as that platform usually doesn't provide feedback for long presses.
+  /// On Android the platform-typical vibration is triggered. On iOS a
+  /// heavy-impact haptic feedback is triggered, alongside the click system
+  /// sound.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/feedback.dart:132]: Nit: "sound, which was observed to be the default behavior on a physical `<iPhone version>` running `<iOS version>`."

File: packages/flutter/lib/src/widgets/feedback.dart [hunk 2]
```diff
@@ -141,6 +142,9 @@ abstract final class Feedback {
       case TargetPlatform.fuchsia:
         return HapticFeedback.vibrate();
       case TargetPlatform.iOS:
+        SystemSound.play(SystemSoundType.click);
+        HapticFeedback.heavyImpact();
+        return Future<void>.value();
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/feedback.dart:147]: Nit: Use [Future.wait](https://main-api.flutter.dev/flutter/dart-async/Future/wait.html) to return a Future that will resolve when both of these Futures have finished.

File: packages/flutter/lib/src/widgets/feedback.dart [hunk 3]
```diff
@@ -151,9 +155,9 @@ abstract final class Feedback {
   /// Wraps a [GestureLongPressCallback] to provide platform specific feedback
   /// for a long press before the provided callback is executed.
   ///
-  /// On Android the platform-typical vibration is triggered. On iOS this
-  /// is a no-op as that platform usually doesn't provide feedback for a long
-  /// press.
+  /// On Android the platform-typical vibration is triggered. On iOS a
+  /// heavy-impact haptic feedback is triggered, alongside the click system
+  /// sound.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/feedback.dart:160]: Probably also include the iOS device and version here.

**Review by justinmc — APPROVED** [review]
LGTM with nits :+1:

**Review by JhonySpark — COMMENTED** [review]
Good solution

**Pull request merged**